src/Components/App/App.js

```diff
@@ -1,26 +1,25 @@
-import './App.css';
-import {fetchAfricanCountries} from "../../apiCalls"
-import { cleanData, getRandomNumber } from '../../utils';
-import { useEffect, useState } from 'react';
-import { Homepage } from '../Homepage/Homepage';
-import { NavBar } from '../NavBar/NavBar';
-import { Routes, Route } from 'react-router-dom';
-import { SavedPage } from '../SavedPage/SavedPage';
-import { ErrorPage } from '../ErrorPage/ErrorPage';
+import "./App.css";
+import { fetchAfricanCountries } from "../../apiCalls";
+import { cleanData, getRandomNumber } from "../../utils";
+import { useEffect, useState } from "react";
+import { Homepage } from "../Homepage/Homepage";
+import { NavBar } from "../NavBar/NavBar";
+import { Routes, Route } from "react-router-dom";
+import { SavedPage } from "../SavedPage/SavedPage";
+import { ErrorPage } from "../ErrorPage/ErrorPage";
 
-function App() {
+const App = () => {
   const [singleFlag, setSingleFlag] = useState([]);
-  const [savedFlags, setSavedFlag] = useState([])
-  const [alert, setAlert] = useState("");
+  const [savedFlags, setSavedFlag] = useState([]);
 
   const addFlag = (newFlag) => {
-    setSavedFlag((prev)=> [...prev, newFlag])
-  }
+    setSavedFlag((prev) => [...prev, newFlag]);
+  };
 
-  const deleteFlag =(name) => {
-   const filteredFlags = savedFlags.filter(flag => flag.name!== name) 
-   setSavedFlag(filteredFlags)
-  }
+  const deleteFlag = (name) => {
```

Comment:
the delete flag function seems to be a bit buggy. i tried saving a bunch of flags for later, and then tried deleting everything, but it doesn't seem to be working as intended. not exactly sure the source of the issue, but it's something to look into.